broker/serializer.go

```diff
@@ -545,8 +482,8 @@ func (sz Serializer) getDateAndTimeValue(dateTimeValue []interface{}) string {
 	var DateTimeValue string
 	if len(dateTimeValue) > 0 {
 		observedAtMap := dateTimeValue[0].(map[string]interface{})
-		if observedAtMap[TYPE] == DATE_TIME {
-			DateTimeValue = observedAtMap[VALUE].(string)
+		if strings.Contains(observedAtMap["@type"].(string), "DateTime") {
```

Comment:
[Major]it should be @value rather than @type.

Reply:
comment incorporated